packages/@aws-cdk/aws-apprunner-alpha/lib/auto-scaling-configuration.ts

```diff
@@ -0,0 +1,181 @@
+   * @attribute
+   */
+  readonly autoScalingConfigurationArn: string;
+
+  /**
+   * The Name of the Auto Scaling Configuration.
+   * @attribute
+   */
+  readonly autoScalingConfigurationName: string;
+
+  /**
+   * The revision of the Auto Scaling Configuration.
+   * @attribute
+   */
+  readonly autoScalingConfigurationRevision: number;
+}
+
+/**
+ * The App Runner Auto Scaling Configuration.
+ *
+ * @resource AWS::AppRunner::AutoScalingConfiguration
+ */
+export class AutoScalingConfiguration extends cdk.Resource implements IAutoScalingConfiguration {
+  /**
+   * Imports an App Runner Auto Scaling Configuration from attributes
+   */
+  public static fromAutoScalingConfigurationAttributes(scope: Construct, id: string,
+    attrs: AutoScalingConfigurationAttributes): IAutoScalingConfiguration {
+    const autoScalingConfigurationArn = attrs.autoScalingConfigurationArn;
+    const autoScalingConfigurationName = attrs.autoScalingConfigurationName;
+    const autoScalingConfigurationRevision = attrs.autoScalingConfigurationRevision;
+
+    class Import extends cdk.Resource {
+      public readonly autoScalingConfigurationArn = autoScalingConfigurationArn
+      public readonly autoScalingConfigurationName = autoScalingConfigurationName
+      public readonly autoScalingConfigurationRevision = autoScalingConfigurationRevision
+    }
+
+    return new Import(scope, id);
+  }
+
+  /**
+   * Imports an App Runner Auto Scaling Configuration from its ARN
+   */
+  public static fromArn(scope: Construct, id: string, autoScalingConfigurationArn: string): IAutoScalingConfiguration {
+    const arn = cdk.Stack.of(scope).splitArn(autoScalingConfigurationArn, cdk.ArnFormat.SLASH_RESOURCE_NAME);
+
+    const resourceParts = arn.resourceName?.split('/');
+
+    if (!resourceParts || resourceParts.length < 3) {
+      throw new Error(`Unexpected ARN format: ${autoScalingConfigurationArn}`);
+    }
+
+    const autoScalingConfigurationName = resourceParts[0];
+    const autoScalingConfigurationRevision = parseInt(resourceParts[1]);
+
+    return AutoScalingConfiguration.fromAutoScalingConfigurationAttributes(scope, id, {
+      autoScalingConfigurationArn,
+      autoScalingConfigurationName,
+      autoScalingConfigurationRevision,
+    });
+  }
+
+  /**
+    * The ARN of the Auto Scaling Configuration.
+    * @attribute
+    */
+  readonly autoScalingConfigurationArn: string;
+
+  /**
+    * The name of the Auto Scaling Configuration.
+    * @attribute
+    */
+  readonly autoScalingConfigurationName: string;
+
+  /**
+   * The revision of the Auto Scaling Configuration.
+   * @attribute
+   */
+  readonly autoScalingConfigurationRevision: number;
+
+  public constructor(scope: Construct, id: string, props: AutoScalingConfigurationProps = {}) {
+    super(scope, id, {
+      physicalName: props.autoScalingConfigurationName,
+    });
+
+    if (props.minSize !== undefined && (props.minSize < 1 || props.minSize > 25)) {
```

Comment:
```suggestion
    if (props.minSize !== undefined && !cdk.Token.isUnresolved(props.minSize) && (props.minSize < 1 || props.minSize > 25)) {
```
We should also verify that the attribute value is resolved before applying validation.